### PR TITLE
Added tests for Cloud Events and verification of reception of events by the Sink

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Added tests for Cloud Events and verification of reception of events by the Sink

**Related issue(s)**

- https://github.com/kyma-project/eventing-manager/issues/74

### CODE REVIEW

**Review by muralov — COMMENTED** [review]
I have found some other improvements not specific to the PR:
* I haven't seen any clean-up for eventing-manager in cleanup_test go files. Do we delete it at the endo file.
* We need to correct the [comment]( https://github.com/mfaizanse/eventing-manager/blob/dfe2d93fbc72c27317162440ea218d38ef8a623e/hack/e2e/eventing/cleanup/cleanup_test.go#L16)
* We need to join errors in test cases so that we show all differences so that we see all at once. You can use errrors.Join

**Review by muralov — COMMENTED** [review]
I have found some other improvements not specific to the PR:
* I haven't seen any clean-up for eventing-manager in cleanup_test go files. Do we delete it at the endo file.
* We need to correct the [comment]( https://github.com/mfaizanse/eventing-manager/blob/dfe2d93fbc72c27317162440ea218d38ef8a623e/hack/e2e/eventing/cleanup/cleanup_test.go#L16)
* We need to join errors in test cases so that we show all differences so that we see all at once. You can use errrors.Join

File: hack/e2e/common/testenvironment/test_environment.go [hunk 1]
```diff
@@ -342,36 +362,115 @@ func (te *TestEnvironment) DeleteSubscriptionFromK8s(name, namespace string) err
...
+		return fmt.Errorf("expected event Type: %s, got event Type: %s", expectedEvent.Type(), gotEvent.Type())
+	}
+
+	originalType, ok := gotEvent.Extensions()[fixtures.EventOriginalTypeHeader]
+	if !ok {
+		return fmt.Errorf("expected event to have header: %s, but its missing", fixtures.EventOriginalTypeHeader)
+	}
+	if expectedEvent.Type() != originalType {
+		return fmt.Errorf("expected originaltype header to have value: %s, but got: %s", expectedEvent.Type(), originalType)
+	}
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply hack/e2e/common/testenvironment/test_environment.go:473]: image

[the remark this answers is not in the document]
Comment (mfaizanse) [reply hack/e2e/common/testenvironment/test_environment.go:473]: updated

File: hack/e2e/common/testenvironment/test_environment.go [hunk 2]
```diff
@@ -342,36 +362,115 @@ func (te *TestEnvironment) DeleteSubscriptionFromK8s(name, namespace string) err
...
+
+	// verify if the event was received.
+	return te.VerifyCloudEventReceivedBySink(expectedCEEvent)
+}
+
+func (te *TestEnvironment) VerifyCloudEventReceivedBySink(expectedEvent cloudevents.Event) error {
+	// define the event
+	gotSinkEvent, err := te.SinkClient.GetEventFromSinkWithRetries(expectedEvent.ID(), Attempts, Interval)
+	if err != nil {
 		te.Logger.Debug(err.Error())
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply hack/e2e/common/testenvironment/test_environment.go:429]: It is printed here so we can see the progress during the multiple retries.

File: hack/e2e/eventing/delivery/delivery_test.go
[the remark this answers is not in the document]
Comment (mfaizanse) [reply hack/e2e/eventing/delivery/delivery_test.go:1]: updated

**Comment by mfaizanse** [comment]
> I have found some other improvements not specific to the PR:
> 
> * I haven't seen any clean-up for eventing-manager in cleanup_test go files. Do we delete it e2e/cleanup/cleanup_test.go?
> * We need to correct the [comment](https://github.com/mfaizanse/eventing-manager/blob/dfe2d93fbc72c27317162440ea218d38ef8a623e/hack/e2e/eventing/cleanup/cleanup_test.go#L16)
> * We need to join errors in test cases so that we show all differences so that we see all at once. You can use errors.Join
> * Are the errors are not displayed twice? Let's run them together and check how to it looks.

We are not deploying `eventing-manager` deployment as part of these tests, so thats why we are not deleting it in the cleanup.

**Review by muralov — APPROVED** [review]

**Pull request merged**